From befe321cf1ad3c19634a4bf770b85f0216c7d764 Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Fri, 25 Oct 2019 01:23:24 +0000
Subject: [PATCH] drop dependency on Openssl md4 support ( temp solution )

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 src/mod_authn_file.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/mod_authn_file.c b/src/mod_authn_file.c
index db1a241..beaa797 100644
--- a/src/mod_authn_file.c
+++ b/src/mod_authn_file.c
@@ -16,7 +16,8 @@
 
 #include "sys-crypto.h"
 #ifdef USE_OPENSSL_CRYPTO
-#include <openssl/md4.h>
+/* temp solution to build against Openssl with MD4 support disabled... */
+#define NO_MD4
 #endif
 
 #include "safe_memclear.h"
-- 
2.23.0

