From 89151699e34d4ebf8747cbd2c4499b41b408860e Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Mon, 20 Dec 2021 00:17:51 -0800
Subject: [PATCH 1/2] use pkg-config to probe libpcre and libpcre2

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 configure.ac | 44 ++++++++++++++------------------------------
 1 file changed, 14 insertions(+), 30 deletions(-)

diff --git a/configure.ac b/configure.ac
index 806347b..fd9d2d0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -913,21 +913,13 @@ AC_ARG_WITH([pcre2],
 )
 AC_MSG_RESULT([$WITH_PCRE2])
 
-if test "$WITH_PCRE2" != no; then
-  if test "$WITH_PCRE2" != yes; then
-    PCRE_LIB="-L$WITH_PCRE2/lib -lpcre2-8"
-    CPPFLAGS="$CPPFLAGS -I$WITH_PCRE/include"
-  else
-    AC_PATH_PROG([PCRE2CONFIG], [pcre2-config])
-    if test -n "$PCRE2CONFIG"; then
-      PCRE_LIB=`"$PCRE2CONFIG" --libs8`
-      CPPFLAGS="$CPPFLAGS `"$PCRE2CONFIG" --cflags`"
-    fi
-  fi
-
-  if test -z "$PCRE_LIB"; then
-    AC_MSG_ERROR([pcre2-config not found, install the pcre2-devel package or build with --without-pcre2])
-  fi
+if test "$WITH_PCRE2" != "no"; then
+  PKG_CHECK_MODULES(PCRE2PKG, [libpcre2-8], [
+    PCRE_LIB=${PCRE2PKG_LIBS}
+    CPPFLAGS="$CPPFLAGS ${PCRE2PKG_CFLAGS}"
+  ], [
+    AC_MSG_ERROR([pcre2 pkgconfig not found, install the pcre2-devel package or build with --without-pcre2])
+  ])
 
   AC_DEFINE([HAVE_PCRE], [1], [libpcre2-8])
   AC_DEFINE([HAVE_PCRE2_H], [1], [pcre.h])
@@ -944,21 +936,13 @@ AC_ARG_WITH([pcre],
 )
 AC_MSG_RESULT([$WITH_PCRE])
 
-if test "$WITH_PCRE" != no && test "$WITH_PCRE2" = "no"; then
-  if test "$WITH_PCRE" != yes; then
-    PCRE_LIB="-L$WITH_PCRE/lib -lpcre"
-    CPPFLAGS="$CPPFLAGS -I$WITH_PCRE/include"
-  else
-    AC_PATH_PROG([PCRECONFIG], [pcre-config])
-    if test -n "$PCRECONFIG"; then
-      PCRE_LIB=`"$PCRECONFIG" --libs`
-      CPPFLAGS="$CPPFLAGS `"$PCRECONFIG" --cflags`"
-    fi
-  fi
-
-  if test -z "$PCRE_LIB"; then
-    AC_MSG_ERROR([pcre-config not found, install the pcre-devel package or build with --without-pcre])
-  fi
+if test "$WITH_PCRE" != "no" && test "$WITH_PCRE2" = "no"; then
+  PKG_CHECK_MODULES(PCREPKG, [libpcre], [
+    PCRE_LIB=${PCREPKG_LIBS}
+    CPPFLAGS="$CPPFLAGS ${PCREPKG_CFLAGS}"
+  ], [
+    AC_MSG_ERROR([pcre pkgconfig not found, install the pcre-devel package or build with --without-pcre])
+  ])
 
   AC_DEFINE([HAVE_PCRE], [1], [libpcre])
   AC_DEFINE([HAVE_PCRE_H], [1], [pcre.h])
-- 
2.24.0

