From 05bdc66922ea660848ffa79c27a828b8a392490b Mon Sep 17 00:00:00 2001
From: Tarun Saha <tsaha@libertyglobal.com>
Date: Wed, 20 Jan 2021 13:33:42 -0800
Subject: [PATCH] client notify prefix expiry event

---
 ClntIfaceMgr/ClntIfaceMgr.cpp | 34 ++++++++++++++++++++++++++++++++--
 ClntIfaceMgr/ClntIfaceMgr.h   |  3 +++
 2 files changed, 35 insertions(+), 2 deletions(-)

diff --git a/ClntIfaceMgr/ClntIfaceMgr.cpp b/ClntIfaceMgr/ClntIfaceMgr.cpp
index fb598c36..52105814 100644
--- a/ClntIfaceMgr/ClntIfaceMgr.cpp
+++ b/ClntIfaceMgr/ClntIfaceMgr.cpp
@@ -453,6 +453,32 @@ int TClntIfaceMgr::numBits(int i) {
     return (bits + 1);
 }
 
+void TClntIfaceMgr::notifyExpireInfo(TNotifyScriptParams& params, SPtr<TIfaceIface> iface,
+                                     SPtr<TIPv6Addr> addr, unsigned int prefixLen,
+                                     TIAType type)
+{
+    stringstream tmp;
+
+    if (iface)
+    {
+        tmp << iface->getID();
+        params.addParam("IFINDEX", tmp.str());
+        params.addParam("IFACE", iface->getName());
+    }
+
+    switch (type)
+    {
+        case IATYPE_IA:
+            params.addAddr(addr, 0, 0);
+            break;
+        case IATYPE_PD:
+            params.addPrefix(addr, prefixLen, 0, 0);
+            break;
+
+    }
+}
+
+
 bool TClntIfaceMgr::modifyPrefix(int iface, SPtr<TIPv6Addr> prefix, int prefixLen,
                                  unsigned int pref, unsigned int valid,
                                  PrefixModifyMode mode,
@@ -621,8 +647,12 @@ bool TClntIfaceMgr::modifyPrefix(int iface, SPtr<TIPv6Addr> prefix, int prefixLe
                                     subprefixLen, pref, valid);
             break;
         case PREFIX_MODIFY_DEL:
-          status = prefix_del( (*i)->getName(), (*i)->getID(), subprefix->getPlain(),
-                               subprefixLen);
+            TNotifyScriptParams expireParams;
+            status = prefix_del( (*i)->getName(), (*i)->getID(), subprefix->getPlain(),
+                                 subprefixLen);
+            notifyExpireInfo(expireParams, *i, subprefix, subprefixLen, IATYPE_PD);
+            TIfaceMgr::notifyScript(ClntCfgMgr().getScript(), "expire", expireParams);
+
             break;
         }
         if (status==LOWLEVEL_NO_ERROR) {
diff --git a/ClntIfaceMgr/ClntIfaceMgr.h b/ClntIfaceMgr/ClntIfaceMgr.h
index ea24751e..d278c002 100644
--- a/ClntIfaceMgr/ClntIfaceMgr.h
+++ b/ClntIfaceMgr/ClntIfaceMgr.h
@@ -90,6 +90,9 @@ public:
                       unsigned int pref, unsigned int valid, PrefixModifyMode mode,
                       TNotifyScriptParams* params /*= NULL*/);
 
+    void notifyExpireInfo(TNotifyScriptParams& params, SPtr<TIfaceIface> iface,
+                      SPtr<TIPv6Addr> addr, unsigned int prefixLen, TIAType type);
+
     std::string XmlFile;
 
     static TClntIfaceMgr* Instance;
-- 
2.24.0

