diff --git a/Messages/Msg.cpp b/Messages/Msg.cpp
index 82870fdb..0836ffd4 100644
--- a/Messages/Msg.cpp
+++ b/Messages/Msg.cpp
@@ -73,8 +73,11 @@ int TMsg::getSize()
     for (opt = Options.begin(); opt!=Options.end(); ++opt)
     {
        Log(Info) << "### CPE Debug - Option with index  " << optionCount++ << LogEnd ;
-       Log(Info) << "### CPE Debug - Option with type " << (*opt)->getOptType() << LogEnd ;
-       pktsize += (*opt)->getSize();
+       if ( (*opt) != NULL)
+       {
+          Log(Info) << "### CPE Debug - Option with type " << (*opt)->getOptType() << LogEnd ;
+          pktsize += (*opt)->getSize();
+       }
     }
     Log(Info) << "### CPE Debug - Packet size of option (Add 4) " << pktsize << LogEnd ;
 
