From 08585f5ed38576626451d9652b800dfb8088cd4e Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Thu, 27 May 2021 14:20:50 -0700
Subject: [PATCH] add support DNS failover for "not implement" error code

---
 src/forward.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/forward.c b/src/forward.c
index 8fb0327..3b99a22 100644
--- a/src/forward.c
+++ b/src/forward.c
@@ -865,7 +865,7 @@ void reply_query(int fd, int family, time_t now)
 
   /* Note: if we send extra options in the EDNS0 header, we can't recreate
      the query from the reply. */
-  if ((RCODE(header) == REFUSED || RCODE(header) == SERVFAIL) &&
+  if ((RCODE(header) == REFUSED || RCODE(header) == SERVFAIL || RCODE(header) == NOTIMP) &&
       forward->forwardall == 0 &&
       !(forward->flags & FREC_HAS_EXTRADATA))
     /* for broken servers, attempt to send to another one. */
-- 
2.24.0

